webapp/src/components/gitlab_issue_selector.tsx

```diff
@@ -25,43 +25,42 @@ interface PropTypes {
     removeValidate: (key: string) => void;
 };
 
-interface StateTypes {
-    invalid: boolean;
-    error: string;
-}
+const GitlabIssueSelector = (props: PropTypes) => {
+    const [invalid, setInvalid] = useState(false);
+    const [error, setError] = useState('')
 
-export default class GitlabIssueSelector extends PureComponent<PropTypes, StateTypes> {
-    constructor(props: PropTypes) {
-        super(props);
-        this.state = {
-            invalid: false,
-            error: '',
-        };
-    }
-
-    componentDidMount() {
-        if (this.props.addValidate && this.props.name) {
-            this.props.addValidate(this.props.name, this.isValid);
+    const isValid = useCallback(() => {                
+        if (!props.required) {
+            return true;
         }
-    }
+        
+        const valid = Boolean(props.value);
+        setInvalid(!valid);
+        return valid;
+    }, [props.value])
 
-    componentWillUnmount() {
-        if (this.props.removeValidate && this.props.name) {
-            this.props.removeValidate(this.props.name);
+    useEffect(() => {
+        return () => {            
+            if (props.removeValidate && props.name) {
+                props.removeValidate(props.name);
+            }
         }
-    }
-
-    componentDidUpdate() {
-        if (this.state.invalid) {
-            this.isValid();
+    }, [])
+    
+    useEffect(() => {
+        if (props.addValidate && props.name) {            
+            props.addValidate(props.name, isValid);
```

Comment:
This code for `componentDidMount` can exist in the same `useEffect` that you used above for `componentWillUnmount`.

Reply:
discussed